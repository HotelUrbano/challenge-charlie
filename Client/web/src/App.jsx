import { memo, useCallback, useEffect, useState } from "react";

import "./Styles/global.css";
import { AppStyle } from "./Styles/app";
import env from "react-dotenv";
import GoogleService from "./Services/google";
import { CardWeather } from "./Components/Card";

export const App = memo(() => {
    const [nameLocale, setNameLocale] = useState(null);
    useEffect(() => {
        async function getLocation() {
            if (!("geolocation" in navigator)) {
                return alert("Navegador nÃ£o compativel com geolocalizaÃ§Ã£o");
            }

            navigator.geolocation.getCurrentPosition(
                async ({ coords: { latitude, longitude } }) => {
                    const {
                        data: { results },
                    } = await GoogleService.getLocationGoogle(
                        latitude,
                        longitude,
                        env.KEY_API
                    );

                    setNameLocale(
                        `${results[6].address_components[0].long_name}, ${results[6].address_components[1].long_name}`
                    );

                    console.log("results", results);
                },
                (error) => {
                    return alert(`Error: ${error}`);
                }
            );
        }

        getLocation();
    }, []);

    const handleChange = useCallback((value) => {}, []);

    const handleOnBlur = useCallback(() => {}, []);
    return (
        <>
            <AppStyle>
                <CardWeather
                    nameLocale={nameLocale}
                    handleChange={handleChange}
                    handleOnBlur={handleOnBlur}
                />
            </AppStyle>
        </>
    );
});
