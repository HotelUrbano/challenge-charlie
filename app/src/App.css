.App .wrapper {
  background-position: center;
  background-size: cover; }

.App .container {
  margin: 0 auto;
  display: block;
  max-width: 600px;
  width: 100%;
  border-radius: 10px;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: center; }

.App .loader {
  background-color: #eeeeee85;
  border-radius: 0;
  margin: 0 auto;
  display: block;
  max-width: 600px;
  width: 100%;
  min-height: 80vh;
  display: flex;
  flex-direction: column;
  justify-content: center; }

.App .weather {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  align-items: center;
  margin-bottom: 10px; }

.App .forecast {
  text-align: center; }
