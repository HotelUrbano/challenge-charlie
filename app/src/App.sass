.App
    .wrapper
        background-position: center
        background-size: cover

    .container
        margin: 0 auto
        display: block
        max-width: 600px
        width: 100%
        border-radius: 10px
        min-height: 100vh
        display: flex
        flex-direction: column
        justify-content: center
    .loader
        background-color: #eeeeee85
        border-radius: 0
        margin: 0 auto
        display: block
        max-width: 600px
        width: 100%
        min-height: 80vh
        display: flex
        flex-direction: column
        justify-content: center
    .weather
        display: flex
        flex-wrap: wrap
        justify-content: space-around
        align-items: center
        margin-bottom: 10px
    .forecast
        text-align: center
