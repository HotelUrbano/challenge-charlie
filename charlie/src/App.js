import React from "react";
import Weather from "../src/components/Weather/Weather";
import Background from '../src/components/Background/Background';


function App() {
  return (
    <div className="App">
        <Background />
          
      <Weather />
    </div>
  );
}

export default App;