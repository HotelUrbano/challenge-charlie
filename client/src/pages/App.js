import React, { Component } from 'react';
import { connect } from 'react-redux';
// Action(s)
import * as actions from '../actions';
import SearchSvg from '../assets/images/search.svg';
// Component
import WeatherCity from '../components/Weather';
import Spinner from '../components/Spinner';

// Para temperaturas abaixo de 15ºC deve ser usado tons de azul,
// para temperaturas acima de 35ºC deve ser usado tons de vermelho
// e use tons de amarelo para as demais temperaturas
class App extends Component {
    state = { inputFocused: false, city: '', convertedTemp: true };

    componentDidMount() {
        this.askForPosition().then(({ latitude, longitude }) => this.props.fetchLatLongWeather(latitude, longitude));
        this.props.fetchBingData();
        this.props.fetchCityWeather();
    }

    askForPosition = () => {
        return new Promise((resolve, reject) => {
            navigator.geolocation.getCurrentPosition(({ coords }) => resolve(coords), error => reject(error));
        });
    };

    handleSubmit = e => {
        e.preventDefault();
        this.props.fetchCityWeather(this.state.city);
    };

    handleConvertTemp = () => {
        this.setState({ convertedTemp: !this.state.convertedTemp });
    };

    render() {
        const { city, loading, bkgImg } = this.props;
        const { convertedTemp } = this.state;

        return (
            <main
                className="main"
                style={{
                    backgroundImage: `url(${bkgImg.image})`
                }}
            >
                <div className={`backdrop ${this.state.inputFocused ? 'backdrop--active' : ''}`} />
                <section className="container home">
                    <h1 className="home__name">Charlie Weather</h1>
                    <form className="search" onSubmit={this.handleSubmit}>
                        <input
                            type="text"
                            className="search__input"
                            onChange={e => this.setState({ city: e.target.value })}
                            placeholder="Search for a city"
                            onFocus={() => this.setState({ inputFocused: true })}
                            onBlur={() => this.setState({ inputFocused: false })}
                        />
                        <button className="search__button" type="submit">
                            <img src={SearchSvg} alt="ButtonSearch" />
                        </button>
                    </form>
                    {loading ? (
                        <Spinner />
                    ) : Object.keys(city).length <= 1 ? (
                        <h1 className="heading-1 error-text">Oops!</h1>
                    ) : (
                        <div className="weather-box">
                            <div className="weather-box__header">
                                <h2 className="heading-2">{city.location.city}</h2>
                                <div className="weather-box__header__group">
                                    <h4 className="heading-4">{city.location.country}</h4>
                                    <h4 className="heading-4">{city.lastBuildDate}</h4>
                                </div>
                            </div>
                            <div className="animate-top">
                                {city.item.forecast.map((day, i) => {
                                    if (i < 3) {
                                        return (
                                            <WeatherCity
                                                key={i}
                                                convertTemp={this.handleConvertTemp}
                                                cityWeather={{ day, city }}
                                                index={i}
                                                tempFormatted={convertedTemp}
                                            />
                                        );
                                    }
                                })}
                            </div>
                        </div>
                    )}
                </section>
            </main>
        );
    }
}

const stateToProps = ({ weather }) => weather;

export default connect(
    stateToProps,
    actions
)(App);
