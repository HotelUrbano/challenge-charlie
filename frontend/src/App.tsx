import React from 'react';

import Layout from '@components/Layout';
import Home from '@pages/Home';

import '@theme/main.scss';

const App: React.FC = () => (
  <Layout>
    <Home />
  </Layout>
);

export default App;

/**
 * @see https://gist.github.com/hcatlin/1027867
 */
