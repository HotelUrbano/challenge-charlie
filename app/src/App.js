import React, { Component } from 'react';
import './App.css';
import axios from 'axios';
import _ from 'lodash';
import SearchBar from './components/SearchBar';

class App extends Component {
    constructor(props) {
        super(props);
        this.state = {
            bgBing: '',
            location: '',
            results: '',
            forecast: [],
            error: ''
        };
    }

    componentDidMount() {
        axios.get('https://bingwallpaper.herokuapp.com').then(res => {
            let bgImage = res.data;
            console.log(res);
            this.setState({
                bgBing: bgImage
            });
        });
        this.getGeoLocation();
    }

    getGeoLocation = () => {
        const findLocation = position => {
            let coordinates =
                position.coords.latitude + ',' + position.coords.longitude;
            let queryLocation = `https://query.yahooapis.com/v1/public/yql?q=select * from weather.forecast where woeid in (SELECT woeid FROM geo.places WHERE text="(${coordinates})")&format=json`;
            axios.get(queryLocation).then(resp => {
                this.setState({
                    location: resp.data.query.results.channel.location.city,
                    forecast: resp.data.query.results.channel.item.forecast
                });
            });
        };
        const errorHandler = error => {
            this.setState({
                error: error
            });
        };

        navigator.geolocation.getCurrentPosition(findLocation, errorHandler);
    };

    inputHandler = e => {
        this.setState({ location: e.target.value });
        this.getByInputLocation();
    };

    getByInputLocation = _.debounce(
        () => {
            let queryLocation = `https://query.yahooapis.com/v1/public/yql?q=select * from weather.forecast where woeid in (SELECT woeid FROM geo.places WHERE text="(${
                this.state.location
            }, BR)")&format=json`;
            axios.get(queryLocation).then(resp => {
                this.setState({
                    results: resp.data.query.results.channel
                });
            });
        },
        1000,
        { maxwait: 1000 }
    );

    render() {
        let sytleBg = {
            backgroundImage: `url(${this.state.bgBing})`,
            width: '100%',
            height: '100vh',
            backgroundPosition: 'center',
            backgroundSize: 'cover'
        };
        return (
            <div className="App" style={sytleBg}>
                <div className="container">
                    <SearchBar
                        location={this.state.location}
                        change={this.inputHandler}
                    />
                </div>
            </div>
        );
    }
}

export default App;
