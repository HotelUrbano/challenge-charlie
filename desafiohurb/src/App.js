import React, { useState, useEffect, useCallback } from "react";
import './App.css';
import { toFahrenheit, toCelsius } from './utils/utils';
import { OpenWeatherService } from './services/openweathermap';
import { OpenCageService } from './services/opencagedata';

function App() {
    const [textInput, setTextInput] = useState('');
    const [coords, setCoords] = useState({
        latitude: null,
        longitude: null
    });
    const [tomorow, setTomorow] = useState();
    const [afterTomorow, setAfterTomorow] = useState();
    const [backgroundImage, setBackgroundImage] = useState();
    const [today, setToday] = useState({});
    const [isCelsius, setIsCelsius] = useState(true);
    const [error, setError] = useState(false);
    const [loading, setLoading] = useState(true);



    useEffect(() => {
        if (!coords.latitude || !coords.longitude) {
            return;
        }
        OpenCageService.getCityByLongitude(coords).then(response => {
            setTextInput(response);
        });
    }, [coords]);

    function changeTemperature() {
        if (isCelsius) {
            setToday({
                ...today,
                temp: toFahrenheit(today.temp).toFixed(0),
            });
            setTomorow(toFahrenheit(tomorow).toFixed(0));
            setAfterTomorow(toFahrenheit(afterTomorow).toFixed(0));
        } else {
            setToday({
                ...today,
                temp: toCelsius(today.temp).toFixed(0),
            });
            setTomorow(toCelsius(tomorow).toFixed(0));
            setAfterTomorow(toCelsius(afterTomorow).toFixed(0));
        }
        setIsCelsius(!isCelsius);
    }

    function handleChange(e) {
        var text = e.target.value;
        setTextInput(text);
    }
    return (
        <div className="App" style={{ backgroundImage: `url(${backgroundImage})` }}>
            <div className="container">


                <section className="search">
                    <input data-icon="(" placeholder="Digite uma cidade..." className="search__input" type="text" value={textInput} onChange={handleChange} />
                </section>
                {error ? (<section className={`error background-none`}><h1>{loading ? "Carregando..." : "Pesquise outra cidade!"}</h1></section>) : <>
                    <section className={`today `}>
                        <div className="today__box">
                            <div className="today__icon">
                                <span>{today.icon}</span>
                            </div>
                            <div className="today__box--data">
                                <div className="temperature">
                                    <span>HOJE</span>
                                    <span className="temperature--action" onClick={changeTemperature} >{today.temp}{isCelsius ? "ºC" : "ºF"}</span>
                                </div>
                                <div className="climate">
                                    <span className="climate__description">{today.description}</span>
                                    <span>Vento:NO {today.wind}km/h</span>
                                    <span>Humidade: {today.humidity}%</span>
                                    <span>Pressão: {today.pressure}hPA</span>
                                </div>
                            </div>
                        </div>
                    </section>

                    <section className={`tomorow`}>
                        <div className="tomorow__text">
                            <span>AMANHÃ</span>
                            <span className="temperature--action" onClick={changeTemperature} >{tomorow}{isCelsius ? "ºC" : "ºF"}</span>
                        </div>
                    </section>

                    <section className={`after-tomorow`}>
                        <div className="after__text">
                            <span>DEPOIS DE AMANHÃ</span>
                            <span className="temperature--action" onClick={changeTemperature}>{afterTomorow}{isCelsius ? "ºC" : "ºF"}</span>
                        </div>
                    </section>
                </>}


            </div>
        </div>
    );
}

export default App;
