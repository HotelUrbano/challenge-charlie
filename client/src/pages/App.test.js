import React from 'react';
import { shallow, mount } from 'enzyme';
import { App } from './App';

describe('App', () => {
    let props = {
        bkgImg: {},
        city: {},
        loading: true
    };
    let app = shallow(<App {...props} />);
    // Smoke Test
    it('should renders properly', () => {
        expect(app).toMatchSnapshot();
    });
    describe('should when mounted', () => {
        const mockGeolocation = {
            getCurrentPosition: jest.fn()
        };
        global.navigator.geolocation = mockGeolocation;
        const mockFetchBingData = jest.fn();
        const mockFetchCityWeather = jest.fn();
        beforeEach(() => {
            props.fetchBingData = mockFetchBingData;
            props.fetchCityWeather = mockFetchCityWeather;
            app = mount(<App {...props} />);
        });
        it('should dispatches the `fetchBingData()` method it receives from props', () => {
            expect(mockFetchBingData).toHaveBeenCalled();
        });
        it('should dispatches the `fetchCityWeather()` method it receives from props', () => {
            expect(mockFetchCityWeather).toHaveBeenCalled();
        });
    });
});
