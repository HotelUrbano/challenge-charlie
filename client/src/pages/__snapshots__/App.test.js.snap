// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    bkgImg={Object {}}
    city={Object {}}
    loading={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="backdrop "
/>,
        <section
          className="container home"
>
          <h1
                    className="home__name"
          >
                    Charlie Weather
          </h1>
          <form
                    className="search"
                    onSubmit={[Function]}
          >
                    <input
                              className="search__input"
                              onBlur={[Function]}
                              onChange={[Function]}
                              onFocus={[Function]}
                              placeholder="Search for a city"
                              type="text"
                    />
                    <button
                              className="search__button"
                              type="submit"
                    >
                              <img
                                        alt="ButtonSearch"
                                        src="search.svg"
                              />
                    </button>
          </form>
          <Spinner />
</section>,
      ],
      "className": "main",
      "style": Object {
        "backgroundImage": "url(undefined)",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "backdrop ",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="home__name"
>
              Charlie Weather
</h1>,
            <form
              className="search"
              onSubmit={[Function]}
>
              <input
                            className="search__input"
                            onBlur={[Function]}
                            onChange={[Function]}
                            onFocus={[Function]}
                            placeholder="Search for a city"
                            type="text"
              />
              <button
                            className="search__button"
                            type="submit"
              >
                            <img
                                          alt="ButtonSearch"
                                          src="search.svg"
                            />
              </button>
</form>,
            <Spinner />,
          ],
          "className": "container home",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Charlie Weather",
              "className": "home__name",
            },
            "ref": null,
            "rendered": "Charlie Weather",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  className="search__input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  placeholder="Search for a city"
                  type="text"
/>,
                <button
                  className="search__button"
                  type="submit"
>
                  <img
                                    alt="ButtonSearch"
                                    src="search.svg"
                  />
</button>,
              ],
              "className": "search",
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "search__input",
                  "onBlur": [Function],
                  "onChange": [Function],
                  "onFocus": [Function],
                  "placeholder": "Search for a city",
                  "type": "text",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <img
                    alt="ButtonSearch"
                    src="search.svg"
/>,
                  "className": "search__button",
                  "type": "submit",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "ButtonSearch",
                    "src": "search.svg",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": "button",
              },
            ],
            "type": "form",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "section",
      },
    ],
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="backdrop "
/>,
          <section
            className="container home"
>
            <h1
                        className="home__name"
            >
                        Charlie Weather
            </h1>
            <form
                        className="search"
                        onSubmit={[Function]}
            >
                        <input
                                    className="search__input"
                                    onBlur={[Function]}
                                    onChange={[Function]}
                                    onFocus={[Function]}
                                    placeholder="Search for a city"
                                    type="text"
                        />
                        <button
                                    className="search__button"
                                    type="submit"
                        >
                                    <img
                                                alt="ButtonSearch"
                                                src="search.svg"
                                    />
                        </button>
            </form>
            <Spinner />
</section>,
        ],
        "className": "main",
        "style": Object {
          "backgroundImage": "url(undefined)",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "backdrop ",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="home__name"
>
                Charlie Weather
</h1>,
              <form
                className="search"
                onSubmit={[Function]}
>
                <input
                                className="search__input"
                                onBlur={[Function]}
                                onChange={[Function]}
                                onFocus={[Function]}
                                placeholder="Search for a city"
                                type="text"
                />
                <button
                                className="search__button"
                                type="submit"
                >
                                <img
                                                alt="ButtonSearch"
                                                src="search.svg"
                                />
                </button>
</form>,
              <Spinner />,
            ],
            "className": "container home",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Charlie Weather",
                "className": "home__name",
              },
              "ref": null,
              "rendered": "Charlie Weather",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    className="search__input"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    placeholder="Search for a city"
                    type="text"
/>,
                  <button
                    className="search__button"
                    type="submit"
>
                    <img
                                        alt="ButtonSearch"
                                        src="search.svg"
                    />
</button>,
                ],
                "className": "search",
                "onSubmit": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "search__input",
                    "onBlur": [Function],
                    "onChange": [Function],
                    "onFocus": [Function],
                    "placeholder": "Search for a city",
                    "type": "text",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <img
                      alt="ButtonSearch"
                      src="search.svg"
/>,
                    "className": "search__button",
                    "type": "submit",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "ButtonSearch",
                      "src": "search.svg",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  "type": "button",
                },
              ],
              "type": "form",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "section",
        },
      ],
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
